lib/names/hostname.go

```diff
@@ -0,0 +1,15 @@
+package names
+
+import (
+	"os"
+	"strings"
+)
+
+// Hostname returns this hosts hostname, converting to lowercase so that
+// it is valid for use in the Calico API.
+func Hostname() (string, error) {
+	if h, err = os.Hostname(); err != nil {
+		return "", err
+	}
+	return strings.ToLower(strings.TrimSpace(h)), nil
```

Comment:
either need this in an else or you need to pull the assignment out of the if statement.

Reply:
whilst this covers immediate use case I'd like to be paranoid and handle things like leading/terminating dashes or dots as part of the upgrade work.